backend/src/index.js

```diff
@@ -0,0 +1,26 @@
+const { ApolloServer } = require('apollo-server');
+const typeDefs = require('./typeDefs');
+const resolvers = require('./resolvers');
+const UsersAPI = require('./datasources/userApi');
+const PostsAPI = require('./datasources/postApi');
+
+
+// DO NOT initialize the endpoint inside the dataSources function!
+// See https://github.com/apollographql/apollo-server/issues/3150
+// Alternatively, set schema.polling.enable to false. (Haven't tested if this works tho)
```

Comment:
No, that doesn't work because it would still create a complete `new` DataSource object.